src/formatting/formatter.rs

```diff
@@ -386,15 +400,64 @@ impl Formatter for FixFormatter {
                 // unit,
                 // icon,
             } => Err(Error::new_format("'fix' formatter is not implemented yet")),
-            Value::Text(_) => Err(Error::new_format(
-                "Text cannot be formatted with 'fix' formatter",
-            )),
-            Value::Icon(_) => Err(Error::new_format(
-                "An icon cannot be formatted with 'fix' formatter",
-            )),
-            Value::Flag => Err(Error::new_format(
-                "A flag cannot be formatted with 'fix' formatter",
-            )),
+            other => Err(Error::new_format(format!(
+                "{} cannot be formatted with 'fix' formatter",
+                other.type_name(),
+            )))
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct DatetimeFormatter {
+    items: Option<Vec<Item<'static>>>,
+}
+
+fn make_static_item(item: Item<'_>) -> Item<'static> {
+    match item {
+        Item::Literal(str) => Item::OwnedLiteral(str.into()),
+        Item::OwnedLiteral(boxed) => Item::OwnedLiteral(boxed),
+        Item::Space(str) => Item::OwnedSpace(str.into()),
+        Item::OwnedSpace(boxed) => Item::OwnedSpace(boxed),
+        Item::Numeric(numeric, pad) => Item::Numeric(numeric, pad),
+        Item::Fixed(fixed) => Item::Fixed(fixed),
+        Item::Error => Item::Error,
+    }
+}
+
+impl DatetimeFormatter {
+    fn new(format: Option<String>, locale: Option<String>) -> Result<Self> {
+        let format = format.as_deref().unwrap_or(DEFAULT_DATETIME_FORMAT);
+
+        let items = match locale.as_deref() {
+            Some(locale) => {
+                let locale = locale.try_into().ok().error("invalid locale")?;
+                StrftimeItems::new_with_locale(format, locale)
+            }
+            None => StrftimeItems::new(format),
+        }
+        .map(make_static_item)
+        .collect();
+
+        Ok(Self { items: Some(items) })
+    }
+}
+
+impl Formatter for DatetimeFormatter {
+    fn format(&self, val: &Value) -> Result<String> {
+        let items = self.items.clone().unwrap();
+        match val {
+            Value::Datetime(timezone) => Ok(match timezone {
+                Some(tz) => Utc::now()
+                    .with_timezone(tz)
+                    .format_with_items(items.iter())
+                    .to_string(),
+                None => Local::now().format_with_items(items.iter()).to_string(),
```

Comment:
I think we should store `DateTime<Utc>` in `Value::Datetime`. It's unlikely that `Datetime` will be ever used to display something other than the current time, but still.